### PR TITLE
feat(ms-inventory): Migrate ms-inventory to EdgeX v3.0

### LINKED ISSUE #129: Migrate ms-inventory to 3.0.1 App SDK
See the following Migration guides.
- https://docs.edgexfoundry.org/3.0/microservices/configuration/V3MigrationCommonConfig/
- https://docs.edgexfoundry.org/3.0/microservices/application/V3Migration/

Other
- Use `Go 1.20` to build and test. 
- Update go.mod to specify `go 1.20`
- Update docker file to use `golang:1.20-alpine3.17` as Builder and alpine:3.17 as base image
- Remove `gcc libc-dev zeromq-dev` from  build and `RUN apk --no-cache add zeromq` from base in Dockerfile
- Change Makefile to use `CGO_ENABLED=0`
- Change copy of config file to be the new YAML file in Dockerfile

### PR DESCRIPTION
**Pull request by seanohair22**
## PR Checklist

- [x] Added label to the Pull Request for easier discoverability and search
- [x] Commit Message meets guidelines as indicated in the URL https://github.com/intel-retail/automated-vending/blob/main/.github/CONTRIBUTING.md
- [x] Every commit is a single defect fix and does not mix feature addition or changes
- [ ] Unit Tests have been added for new changes
- [ ] Updated Documentation as relevant to the changes
- [x] All commented out code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with repository license and clearly outlined the added dependency.
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide details below)

## Issue this PR will close

close: #**129**

closes #129

## Test Instructions if applicable

- Comment out lines, 8-31, lines 58 - 238 and line 240 in `docker-compose.ac.yml` (so that only ms-inventory service and inventory volume are not commented out)
- Run `make docker` in ms-inventory dir
- Run `make run` in automated-vending dir
- Run `make run-portainer` and check ms-inventory logs

### CODE REVIEW

**Review by brian-intel — APPROVED** [review]

File: ms-inventory/Makefile [hunk 1]
```diff
@@ -19,7 +19,7 @@ docker:
 		.
 
 gobuild: tidy
-	CGO_ENABLED=1 GOOS=linux go build -ldflags='-s -w' -a -installsuffix cgo main.go
+	CGO_ENABLED=0 GOOS=linux go build -ldflags='-s -w' -a main.go
```
Comment (jim-wang-intel) [inline ms-inventory/Makefile:22]: do we still ldflags if no cgo?

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: ms-inventory/res/configuration.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Copyright © 2023 Intel Corporation. All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+
+Writable:
+  LogLevel: INFO
+
+Service:
+  Host: localhost
+  Port: 48095
+  StartupMsg: This microservice exposes a CRUD interface for an inventory of items in a Automated Checkout
```
Comment (ejlee3) [inline ms-inventory/res/configuration.yaml:10]:
```suggestion
  StartupMsg: This microservice exposes a CRUD interface for an inventory of items in an Automated Checkout System
```

**Review by ejlee3 — CHANGES_REQUESTED** [review]

File: ms-inventory/res/configuration.yaml (hunk 2 above)
Comment (lenny-intel) [reply ms-inventory/res/configuration.yaml:10]: @seanohair22 , @ejlee3 , @brian-intel , we should change all occurrences of `Automated Checkout` to `Automated Vending` in the code and messages.

**Review by lenny-intel — COMMENTED** [review]
Need to change all instances of `automated-checkout` to `automated-vending`

File: ms-inventory/Dockerfile [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
 
 LABEL license='SPDX-License-Identifier: BSD-3-Clause' \
-  copyright='Copyright (c) 2022: Intel'
+  copyright='Copyright (c) 2023: Intel'
 
 
 # add git for go modules
-RUN apk update && apk add --no-cache make git gcc libc-dev zeromq-dev linux-headers
+# hadolint ignore=DL3018
+RUN apk update && apk add --no-cache make git linux-headers
```
[the remark this answers is not in the document]
Comment (seanohair22) [reply ms-inventory/Dockerfile:12]: It's not needed and should be removed or is needed and should be kept?

**Review by lenny-intel — APPROVED** [review]
LGTM

File: ms-inventory/Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (lenny-intel) [reply ms-inventory/Dockerfile:12]: Remove

**Review by lenny-intel — APPROVED** [review]
LGTM

**Review by ejlee3 — APPROVED** [review]
LGTM

**Review by jim-wang-intel — APPROVED** [review]

**Pull request merged**